www/js/services.js

```diff
@@ -0,0 +1,38 @@
+angular.module('open311.services', [])
+.factory('API', function($http, $httpParamSerializer, $q) {
+
+    var recentCasesData = [];
+
+    var getRequests = function(lat, long) {
+      var deferred = $q.defer();
+
+      var params = $httpParamSerializer({
+          lat: lat,
+          long: long 
+      });
+
+      $http.get('https://seeclickfix.com/open311/v2/requests.json?' + params)
+        .then(function(response) {
+          recentCasesData = response.data;
+          deferred.resolve(response);
+        }, function(response) {
+          deferred.reject(response);
+        });
+      
+      return deferred.promise;
+    };
+
+    var getCase = function(caseID) {
+      for (var i = 0; i < recentCasesData.length; i++) {
+        if (recentCasesData[i].service_request_id === parseInt(caseID)) {
+          return recentCasesData[i];
+        }
+      }
+      return null;
```

Comment:
`getCase` requires `getRequests` to be called to populate `recentCasesData` before it can be used.

This can cause a problem if someone wants to go to a details page (for instance, through a push notification) before going to a list view of all of the requests.

Can you replace this with its own API call?  If you want to get the data for one specific case, you can hit a URL like this:

https://seeclickfix.com/open311/v2/requests/[service_request_id].json

For example, to hit the request with `service_request_id` of 2278001, you would go to this URL:

https://seeclickfix.com/open311/v2/requests/2278001.json

Let me know if that makes sense.